libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -209,6 +223,11 @@ - (SFSyncState*) syncDownWithTarget:(SFSyncDownTarget*)target options:(SFSyncOpt
 /** Resync
  */
 - (SFSyncState*) reSync:(NSNumber*)syncId updateBlock:(SFSyncSyncManagerUpdateBlock)updateBlock {
+    if (syncId.integerValue == self.syncIdRunning) {
```

Comment:
This change is not introducing this constraint. Currently all syncs are dispatched through a serial queue and therefore there is only one running at given time (for a given store).
If we relax that constraint, we will have to at least prevent two concurrent syncs to/from the same soup. And then keeping track of the collection of running syncs like you suggested will be needed.